crates/context/config/src/client/config.rs

```diff
@@ -56,6 +56,22 @@ pub struct Credentials {
     pub secret_key: SecretKey,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(try_from = "sn_serde_creds::Credentials")]
+pub struct SnCredentials {
+    pub account_id: String,
+    pub public_key: String,
+    pub secret_key: String,
+}
+
+#[non_exhaustive]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum CryptoCredentials {
+    Near(Credentials),
+    Starknet(SnCredentials),
+}
```

Comment:
try localizing credentials to the protocol modules

```suggestion
pub enum Credentials {
    Near(near::Credentials),
    Starknet(starknet::Credentials),
}
```

Reply:
Updated and localized